src/services/services.ts

```diff
@@ -1558,14 +1558,14 @@ namespace ts {
                 options.triggerCharacter);
         }
 
-        function getCompletionEntryDetails(fileName: string, position: number, name: string, formattingOptions: FormatCodeSettings | undefined, source: string | undefined, preferences: UserPreferences = emptyOptions): CompletionEntryDetails | undefined {
+        function getCompletionEntryDetails(fileName: string, position: number, name: string, formattingOptions: FormatCodeSettings | undefined, source: string | undefined, preferences: UserPreferences = emptyOptions, data?: CompletionEntryData): CompletionEntryDetails | undefined {

```

Comment:
I think this should be `data: unknown` here or or in server you would need to convert it to this type with verification of type?

Reply:
Yeah, verifying in the server is probably a good idea. I’ll add that.